mysite/w6/beanstalk.py

```diff
@@ -1,140 +1,146 @@
+from consts import gfood_codes
 from models.models import AdviceSection, Advice, AdviceGroup
-from utils.data_formatting import safe_loads
 from utils.text_formatting import getItemDisplayName
 from flask import g as session_data
 
 TEN_K = 1
 HUNNIT_K = 2
 
+BEANSTACK_GOAL = 10**4
+SUPER_BEANSTACK_GOAL = 10**5
 
-def __get_ninja_section(raw):
-    default = "[]"
-    return safe_loads(raw.get("Ninja", default))
-
-
-def __get_beanstalk_data(raw):
-    if ninja_section := __get_ninja_section(raw):
-        return ninja_section[-4]
-    else:
-        return []
 
 def section_beanstalk():
-    beanstalk_bought = "Gold Food Beanstalk" in session_data.account.jade_emporium_purchases
     upgrade_bought = "Supersized Gold Beanstacking" in session_data.account.jade_emporium_purchases
-    beanstalk_data = __get_beanstalk_data(session_data.account.raw_data)
 
-    if not (beanstalk_bought or beanstalk_data):
+    if "Gold Food Beanstalk" not in session_data.account.jade_emporium_purchases:
         return AdviceSection(
             name="Giant Beanstalk",
             tier="",
             header="Come back once you've bought the \"Gold Food Beanstalk\" from the Jade Emporium",
             picture="Jade_Vendor.gif",
-            collapse=False
+            collapse=True,
         )
 
-    gfood_codes = ["PeanutG", "ButterBar", *[f"FoodG{i}" for i in range(1, 14)]]
     gold_foods = dict.fromkeys(gfood_codes, 0)
-    beanstalk_status = dict(zip(gfood_codes, beanstalk_data))
 
+    #Assets contains totals from Storage and inventories
     for gfood in gfood_codes:
         gold_foods[gfood] += session_data.account.assets.get(gfood).amount
```

Comment:
if `.get()` ends up being `None` then it's going to error out when trying to do `None.amount`

Reply:
Covered already, no worries. Trying to .get() on an Asset that doesn't exist will return a new Asset with amount of 0. https://github.com/TwoSpookyBoos/IdleOnAutoReviewBot/blob/455ea007713516827e10a65c261208c00e8b0726/mysite/models/models.py#L543